Enigma machine (#59)
infinityworksltd/engineering: _posts/2019-09-24-enigma-machine.md

---
layout: post
title: Enigma Machines
author: flock3
---

![The first presentation slide showing the infintiy works logo]({{ "/images/enigma/0001.jpg"}})

# A short Center Parcs story of Enigma

At the annual InfinityWorks conference every year, we try and bring something new and interesting 'to the stage' (literally) and this year, given how much we'd grown, we realised that the standard trope of the directors standing on the stage and using our annual conference as a golden opportunity to mock various members of the organisation just simply doesn't scale to a room of 200 people.

> We decided therefore that this year, we should make our conference an actual conference.  We had 16 speakers with various masters of ceremonies to introduce them, culminating in a spectacular talk from Jamie Andrew about the challanges he faced in coping with challange and change. 

Reviewer: `s/challange/challenge/`

> The theme of the conference was therefore suitably: "Embracing Change and Ambiguity".  The conference started off with a bag, with our very own CTO Dan Rathbone giving an excellent keynote about the alternative history of InfintiyWorks and how we started, leading now to a brand new InfinityWorks meme for "Seizing the Fish" (perhaps you had to be there).

Reviewer: s/bag/bang/

> The theme of the conference was therefore suitably: "Embracing Change and Ambiguity".  The conference started off with a bag, with our very own CTO Dan Rathbone giving an excellent keynote about the alternative history of InfintiyWorks and how we started, leading now to a brand new InfinityWorks meme for "Seizing the Fish" (perhaps you had to be there).

Author: ```suggestion
The theme of the conference was therefore suitably: "Embracing Change and Ambiguity".  The conference started off with a bang, with our very own CTO Dan Rathbone giving an excellent keynote about the alternative history of InfintiyWorks and how we started, leading now to a brand new InfinityWorks meme for "Seizing the Fish" (perhaps you had to be there).
```

I was given the opportunity to speak at this years conference, and given my general desire to push the boat out, I thought rather than deliver a talk about some technical or client challange, I would give a talk about a subject I knew almost nothing about - requiring me to learn a great deal in the 2 weeks before the conference!

![A WW2 poster with the caption 'careless talk costs lives']({{ "/images/enigma/0002.jpg"}})

> My talk started with a startling shout of "Careless talk costs lives" - I absoloutely love these posters, and truly belive we should frame them up around the office.

Reviewer: s/belive/believe/

> My talk started with a startling shout of "Careless talk costs lives" - I absoloutely love these posters, and truly belive we should frame them up around the office.

Author: ```suggestion
My talk started with a startling shout of "Careless talk costs lives" - I absolutely love these posters and truly believe we should frame them up around the office.
```

> The story begins with a trip to Beltchley park I made with my wife our (then) 6 month old daughter and my brother and father. 

Reviewer: s/Beltchley/Bletchley/

> The story begins with a trip to Beltchley park I made with my wife our (then) 6 month old daughter and my brother and father. 

Author: ```suggestion
The story begins with a trip to Bletchley park I made with my wife our (then) 6-month-old daughter and my brother and father. 
```

![An image with the caption 'I'm going to tell you how enigma works']({{ "/images/enigma/0006.jpg"}})

Just as the title slide says, let's get into how enigma works.

![A picture of an Enigma machine with the keyboard, lamp board, reflectors, rotors and plug board shown]({{ "/images/enigma/0011jpg"}})

> This then is the fundamental overview of an enigma and the components therin

Reviewer: s/enigma/Enigma/

> This then is the fundamental overview of an enigma and the components therin

Author: ```suggestion
This then is the fundamental overview of an Enigma and the components therein
```

We decided therefore that this year, we should make our conference an actual conference.  We had 16 speakers with various masters of ceremonies to introduce them, culminating in a spectacular talk from Jamie Andrew about the challanges he faced in coping with challange and change. 

> The theme of the conference was therefore suitably: "Embracing Change and Ambiguity".  The conference started off with a bang, with our very own CTO Dan Rathbone giving an excellent keynote about the alternative history of InfintiyWorks and how we started, leading now to a brand new InfinityWorks meme for "Seizing the Fish" (perhaps you had to be there).

Reviewer: `s/InfintiyWorks/InfinityWorks/`  (sorry, I don't have edit rights)

> I was given the opportunity to speak at this years conference, and given my general desire to push the boat out, I thought rather than deliver a talk about some technical or client challange, I would give a talk about a subject I knew almost nothing about - requiring me to learn a great deal in the 2 weeks before the conference!

Reviewer: `s/challange/challenge/`
